src/components/ArmyBuildPane.jsx

```diff
@@ -17,30 +17,42 @@ const Container = styled.div`
 `;
 
 const ArmyBuildPane = () => {
+
   const [unitList, setUnitList] = useState({});
   const [listIndex, setListIndex] = useState(1);
 
   const addUnit = unit => {
+    unit.selected_upgrades = {};
     setUnitList({
       ...unitList,
       [listIndex]: unit
     });
     setListIndex(listIndex + 1);
   };
+
   const removeUnit = index => {
-    const {[index]: removedValue, ...remaining} = unitList;
-    setUnitList(
-      remaining
-    );
+    const { [index]: removedValue, ...remaining } = unitList;
+    setUnitList(remaining);
+  };
+
+  const addUpgrade = (unitKey, upgradeKey, upgrade) => {
+      let unitListCpy = {...unitList};
+      unitListCpy[unitKey].selected_upgrades[upgradeKey] = upgrade;
+      setUnitList(unitListCpy);
   };
+
   return (
     <Container>
       <UsageBar usage={calculateUsage(unitList)} />
       <SelectionPane
         addFunction={addUnit}
         factionUnits={getFaction(FACTIONS.REBEL)}
       />
-      <ArmyList removeFunction={removeUnit} factionUnits={unitList} />
+      <ArmyList
```

Comment:
Pro-tip: You can use whitespace in JXS to visually break up unrelated elements inside of rendering. Not necessarily to be applied here because it's a pretty simple component (though it wouldn't hurt!) but if there are components that aren't directly related being rendered together, I'd usually include line breaks to help split them up visually, especially when they have long multi-line prop lists like this. Easier to see where things start/end.